### PR TITLE
Pyupgrade cleanup

### PR DESCRIPTION
**Pull request by imwatsi**
### What was wrong?

A `pyupgrade` does not adequately upgrade syntax. A manual approach is required. Issue #1814 

### How was it fixed?

I manually went through the codebase, upgrading formatted strings to f-strings. Other cleanup actions are to be taken, f-strings were the initial focus.

### To-Do

- [X] F-strings
- [X] Other syntax upgrades

### CODE REVIEW

File: eth/_utils/rlp.py [hunk 1]
```diff
@@ -71,29 +71,20 @@ def validate_rlp_equal(obj_a: BaseBlock,
     diff = diff_rlp_object(obj_a, obj_b)
     if len(diff) == 0:
         raise TypeError(
-            "{} ({!r}) != {} ({!r}) but got an empty diff".format(
-                obj_a_name,
-                obj_a,
-                obj_b_name,
-                obj_b,
-            )
+            f"{obj_a_name} ({repr(obj_a)}) != "
```
Comment (pipermerriam) [inline eth/_utils/rlp.py]: These two are equivalent:

```
f"{repr(obj)}"
f"{obj!r}"
```

can you update to use the `!r` syntax?

File: eth/_utils/version.py [hunk 2]
```diff
@@ -9,10 +9,10 @@ def construct_evm_runtime_identifier() -> str:
...
-        __version__,
-        platform=sys.platform,
-        v=sys.version_info,
-        # mypy Doesn't recognize the `sys` module as having an `implementation` attribute.
-        imp=sys.implementation,
-    )
+    
+    platform = sys.platform
+    v = sys.version_info
+    # mypy Doesn't recognize the `sys` module as having an `implementation` attribute.
```
Comment (pipermerriam) [inline eth/_utils/version.py]: looks like this comment can be removed.

**Comment by pipermerriam** [comment]
Take a look at the "files" part of the diff, a few of the files had permissions changed.

File: eth/_utils/rlp.py (hunk 1 above)
Comment (imwatsi) [reply eth/_utils/rlp.py]: I had no idea I could do that :) That definitely looks much better. I will update.

**Comment by imwatsi** [comment]
I have made the changes you requested:

- Revert file permissions
- Use !r syntax

I also fixed pylint test errors.

*P.S. My apologies for not waiting for approval on Gitcoin before working on this.*

**Comment by imwatsi** [comment]
Ran a final pyupgrade just now, in the last commit.

**Pull request merged**